### PR TITLE
Added logic for showing notification on risk level change (EXPOSUREAPP-2569)

### PR DESCRIPTION
**Pull request by SamuraiKek**
### Description

This PR introduces a new dialog which informs the user what to do when the risk level lowers from red to green.
### Steps to reproduce

Due to the nature of how the risk is calculated, there is currently no easy way of testing this.
Prerequisites include:
- Having two android phones, staying close to each other, both having ENF active.
- One of which needs to get a positive result.
- From the phone with the positive result, submit the keys to the server.
- Once the keys from the first phone have been downloaded and the risk calculated, wait for the risk status on the second phone to change to high, 
- Reset the risk level on the second phone by going into the testers menu.
- With the risk level reset, the notification should pop up

### CODE REVIEW

File: Corona-Warn-App/src/main/res/values/strings.xml
Comment (d4rken) [inline Corona-Warn-App/src/main/res/values/strings.xml:76]: Don't use strings.xml for preference keys add a static string variable to `LocalData`

File: Corona-Warn-App/src/main/res/values-de/strings.xml
Comment (d4rken) [inline Corona-Warn-App/src/main/res/values-de/strings.xml:381]: ` \n ` Spaces after and before newlines?

`2f`?

Sure that the string is correct?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt:184]: The viewmodel should decide whether the dialog needs to be shown, preferably via `SingleLiveEvent`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt:361]: "whether", not :partly_sunny:

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt:366]: Since this is only used for the dialog, let name this `isUserToBeNotifiedOfLoweredRiskLevel` so no one accidentally uses this data for something else.

**Review by d4rken — CHANGES_REQUESTED** [review]

File: Corona-Warn-App/src/main/res/values-de/strings.xml
Comment (janetback) [inline Corona-Warn-App/src/main/res/values-de/strings.xml:381]: Just checking - is '\n 2f.' ok here?

**Review by janetback — CHANGES_REQUESTED** [review]

File: Corona-Warn-App/src/main/res/values-de/strings.xml
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/res/values-de/strings.xml:381]: No, it's not. It's a typo. I'm getting it fixed

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt]: I always mix these two up.  It's fixed now

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt]: Yeah, you're right,  I've changed it.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt:187]: Ok, the showing of the notification is now handled by the viewmodel.

File: Corona-Warn-App/src/main/res/values-de/strings.xml
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/res/values-de/strings.xml]: I've removed the spaces and the typo.

File: Corona-Warn-App/src/main/res/values/strings.xml
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/res/values/strings.xml]: Ok, I've implemented a static string in LocalData and removed the string from strings.xml

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt:187]: The logic and decisions are still being made in the Fragment and not the ViewModel, it would also be good to not require a manual refresh call. Maybe we can have call.

**Review by janetback — DISMISSED** [review]
UA approved

**Review by d4rken — DISMISSED** [review]
lgtm

**Review by harambasicluka — DISMISSED** [review]

**Review by janetback — DISMISSED** [review]
UA approved

**Review by d4rken — DISMISSED** [review]

**Review by d4rken — APPROVED** [review]
lgtm if tests pass

**Review by harambasicluka — APPROVED** [review]

**Pull request merged**
